spirv_msl.cpp

```diff
@@ -1336,6 +1394,8 @@ uint32_t CompilerMSL::get_metal_resource_index(SPIRVariable &var, SPIRType::Base
 	switch (basetype)
 	{
 	case SPIRType::Struct:
+		if (execution.model == ExecutionModelVertex && next_metal_resource_index.msl_buffer == 0)
+			next_metal_resource_index.msl_buffer = 1;
```

Comment:
This is a hard-coded special case that will break in the general sense. What is the issue you are trying to fix with this change?

See also the MSLResourceBinding API struct for the general mechanism for configuring resource indexes.